internal/driver/readhandler.go

```diff
@@ -103,7 +105,26 @@ func makeReadRequest(deviceClient *opcua.Client, req sdkModel.CommandRequest) (*
 
 	// make new result
 	reading := resp.Results[0].Value.Value()
-	return newResult(req, reading)
+	result, err := newResult(req, reading)
+
+	// Try to get the sourcetime timestamp from the result since it is the closest timestamp to the event source. If not applicable, use servertime or the current time.
+	if err == nil {
```

Comment:
To reduce the nesting level the error check and handling should here better be performed with inverted condition and possible early return from function in the style:

```suggestion
	if err != nil {
	    return nil, err
	    // or possibly more detailed:
	    // return nil, fmt.Error(...)
	}
```

That's also more in line the other go code is structured.

Reply:
done